### DISCUSSION TITLE
Cypress test unable to run with nx

### DISCUSSION THREAD

**Discussion by shivanshsingh-sprih** [cypress-io/cypress#26649] [Questions and Help]
Here is my package.json file:
 
{
  "name": "gulmohar",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "start": "nx serve",
    "build": "nx build",
    "test": "nx test",
    "cy:open:mock": "npx nx e2e adminapp-e2e --env.environment=mock --watch",
    "cy:open:live": "npx nx e2e adminapp-e2e --env.environment=live --watch",
    "lint": "nx lint",
    "prettier:fix": "nx format:write",
    "prettier:check": "nx format:check",
    "storybook": "nx run components:storybook || npx tailwindcss -i apps/adminapp/src/app/tailwind.css -o ./tailwind.css --watch"
  },
  "private": true,
  "dependencies": {
    "@emotion/react": "^11.10.5",
    "@mantine/carousel": "^5.10.5",
    "@mantine/core": "^5.10.5",
    "@mantine/dates": "^5.10.5",
    "@mantine/dropzone": "^5.10.5",
    "@mantine/form": "^5.10.5",
    "@mantine/hooks": "^5.10.5",
    "@mantine/notifications": "^5.10.5",
    "@reduxjs/toolkit": "^1.9.1",
    "@tanstack/react-table": "^8.2.3",
    "ag-grid-community": "^28.2.1",
    "ag-grid-react": "^28.2.1",
    "axios": "^0.27.2",
    "chart.js": "^3.8.0",
    "classnames": "^2.3.2",
    "config": "^3.3.7",
    "core-js": "^3.6.5",
    "dayjs": "^1.11.7",
    "express": "4.17.2",
    "file-loader": "^6.2.0",
    "fs": "^0.0.1-security",
    "js-file-download": "^0.4.12",
    "jscpd": "^3.5.1",
    "path": "^0.12.7",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "react-intl": "^6.0.3",
    "react-loading-skeleton": "^3.1.0",
    "react-promise-tracker": "^2.1.1",
    "react-redux": "^8.0.5",
    "react-router-dom": "^6.6.2",
    "react-scrollbar": "^0.5.6",
    "recharts": "^2.1.10",
    "regenerator-runtime": "0.13.7",
    "tabler-icons-react": "^1.48.0",
    "tslib": "^2.3.0",
    "use-react-router-breadcrumbs": "^3.2.1",
    "util": "^0.12.4",
    "webfontloader": "^1.6.28"
  },
  "devDependencies": {
    "@babel/core": "7.12.13",
    "@babel/preset-typescript": "7.12.13",
    "@nrwl/cli": "^15.9.2",
    "@nrwl/cypress": "^15.9.2",
    "@nrwl/eslint-plugin-nx": "14.1.5",
    "@nrwl/express": "^14.1.5",
    "@nrwl/jest": "14.1.5",
    "@nrwl/linter": "14.1.5",
    "@nrwl/node": "14.1.5",
    "@nrwl/nx-cloud": "14.0.3",
    "@nrwl/react": "14.1.5",
    "@nrwl/storybook": "14.1.5",
    "@nrwl/web": "14.1.5",
    "@nrwl/workspace": "14.1.5",
    "@storybook/addon-essentials": "~6.4.12",
    "@storybook/builder-webpack5": "~6.4.12",
    "@storybook/core-server": "~6.4.12",
    "@storybook/manager-webpack5": "~6.4.12",
    "@storybook/react": "~6.4.12",
    "@svgr/webpack": "^5.4.0",
    "@testing-library/react": "13.1.1",
    "@types/express": "4.17.13",
    "@types/jest": "27.4.1",
    "@types/node": "16.11.7",
    "@types/react": "18.0.8",
    "@types/react-dom": "18.0.3",
    "@types/react-router-dom": "5.3.3",
    "@typescript-eslint/eslint-plugin": "~5.18.0",
    "@typescript-eslint/parser": "~5.18.0",
    "axios-mock-adapter": "^1.21.1",
    "babel-jest": "27.5.1",
    "babel-loader": "8.1.0",
    "cypress": "^12.10.0",
    "eslint": "~8.12.0",
    "eslint-config-prettier": "8.1.0",
    "eslint-plugin-cypress": "^2.10.3",
    "eslint-plugin-import": "2.26.0",
    "eslint-plugin-jsx-a11y": "6.5.1",
    "eslint-plugin-react": "7.29.4",
    "eslint-plugin-react-hooks": "4.5.0",
    "jest": "27.5.1",
    "nx": "14.1.5",
    "postcss-loader": "^7.0.2",
    "prettier": "^2.5.1",
    "react-test-renderer": "18.1.0",
    "tailwindcss": "^3.2.6",
    "ts-jest": "27.1.4",
    "ts-node": "9.1.1",
    "typescript": "~4.6.2",
    "url-loader": "^3.0.0",
    "webpack": "^5.75.0",
    "yarn": "^1.22.19"
  }
}

after npm run cy:open:live => this error is occuring ->

Invalid source or target versions. Source: 5.0, Target: 5.1.

Only backwards compatibility between "5.1" and "5.0" is supported.
This error can be caused by "@nrwl/..." packages getting out of sync or outdated project graph cache.
Check the versions running "nx report" and/or remove your "nxdeps.json" file (in node_modules/.cache/nx folder).

I have tried npm cache clean --force and deleting node_modules folder but didn't work. Can anyone help me to fix this?

**Comment by portyu9** [comment]
Hi @shivanshsingh-sprih 

The error is coming from **Nx package versions being out of sync**, not from Cypress.

Your dependencies include a mixture such as:

```text
@nrwl/cli       15.9.2
@nrwl/cypress   15.9.2

nx              14.1.5
@nrwl/...       14.1.5
```

That matches the error itself:

```text
Invalid source or target versions.
Source: 5.0, Target: 5.1.

This error can be caused by "@nrwl/..." packages getting out of sync
```

Deleting `node_modules` or cleaning the npm cache will simply reinstall the same incompatible version combination.

First confirm it:

```bash
npx nx report
```

Then choose **one Nx version line** for the workspace.

If this branch should remain on Nx 14, downgrade the 15.9.2 Nx packages to the matching 14.x version.

If you intended to upgrade to Nx 15, use Nx migrations rather than manually changing two packages. For the old workspace that would be along the lines of:

```bash
npx nx migrate @nrwl/workspace@15.9.2
npm install
npx nx migrate --run-migrations
```

Then clear Nx's own cached project information if necessary:

```bash
npx nx reset
```

The important part is that `nx` and the official `@nrwl/*` Nx packages involved in the workspace should be on compatible/matching versions.

For current Nx projects the package namespace has largely moved from `@nrwl/*` to `@nx/*`, but the same rule still applies: Nx explicitly recommends keeping `nx` and official Nx plugin versions synchronized and using `nx migrate` to update them.

So I would fix the Nx dependency graph first; changing Cypress configuration is not the solution to this particular error.

There were no screenshots or comments in the discussion that suggest a different cause.

Please mark this answer as accepted if it helped, thank you.
